app/controllers/streams_controller.rb

```diff
@@ -10,20 +10,29 @@ def show
     response.headers['Content-Type'] = 'text/event-stream'
     response.headers['Cache-Control'] = 'no-cache'
 
-    @job = Job.find(params[:id])
-    @execution = JobExecution.find_by_id(@job.id)
+    return response.stream.close unless job.active? && execution
 
-    streamer = EventStreamer.new(response.stream, &method(:event_handler))
-
-    return response.stream.close unless @job.active? && @execution
-
-    @execution.viewers.push(current_user)
+    execution.viewers.push(current_user)
     ActiveRecord::Base.clear_active_connections!
-    streamer.start(@execution.output)
+
+    sleep(0.1) until execution.active?
```

Comment:
Is there any scenario where this could end up as an infinite loop?  Like if the `JobExecution` was canceled or removed from the queue?  (I don't know if that's possible right now, but it seems like a reasonable enhancement.)

Or if the thread starts and ends in less than 1/10 of a second?

Maybe something more like `sleep(0.1) while execution.pending?`